### ISSUE TITLE
AppCrash when using NativeChildren for navigation via decompose

### ISSUE THREAD

**Issue by martin-reiterer** [alexzhirkevich/compose-cupertino#35]
When using `NativeChildren` (part of `cupertino-decompose` module) for using UIKit based Navigation approach, an AppCrash can be observed in for several consecutive navigation events (iOS only). The AppCrash does not always occur at the same time. Sometimes a problem already occurs with the second navigation action, but it can also happen that you have to navigate much longer. 

This behavior can also be reproduced with the sample app provided in the `shared` module. Decompose throws an error, because a navigation key is pushed several times in the same navigation stack. In the main navigation stack, which is initialized in the sample app, no multiple occurrences of a navigation key are recognizable. Is it possible that this is a state/concurrency problem that happens during recomposition?

The following stack grace is logged in the case of an AppCrash:

```
Uncaught Kotlin exception: kotlin.IllegalArgumentException: Key DefaultRootComponent.Config.Icons_-f2e08u was used multiple times 
    at 0   shared                              0x109d98a77        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 119 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:28:37)
    at 1   shared                              0x109d91fe3        kfun:kotlin.Exception#<init>(kotlin.String?){} + 115 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:23:44)
    at 2   shared                              0x109d92203        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 115 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:34:44)
    at 3   shared                              0x109d92583        kfun:kotlin.IllegalArgumentException#<init>(kotlin.String?){} + 115 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:59:44)
    at 4   shared                              0x10a84f4bf        kfun:androidx.compose.runtime.saveable.SaveableStateHolderImpl.SaveableStateProvider$lambda$0#internal + 911 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime-saveable/src/commonMain/kotlin/androidx/compose/runtime/saveable/SaveableStateHolder.kt:89:89)
    at 5   shared                              0x10a84fa27        kfun:androidx.compose.runtime.saveable.SaveableStateHolderImpl.$SaveableStateProvider$lambda$0$FUNCTION_REFERENCE$1.invoke#internal + 115 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime-saveable/src/commonMain/kotlin/androidx/compose/runtime/saveable/SaveableStateHolder.kt:88:13)
    at 6   shared                              0x109eb8047        kfun:kotlin.Function1#invoke(1:0){}1:1-trampoline + 107 (/Users/teamcity/.gradle/daemon/8.2.1/[K][Suspend]Functions:1:1)
    at 7   shared                              0x10a0201c7        kfun:androidx.compose.runtime.DisposableEffectImpl.onRemembered#internal + 183 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Effects.kt:82:51)
    at 8   shared                              0x10a125dd3        kfun:androidx.compose.runtime.RememberObserver#onRemembered(){}-trampoline + 91 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/RememberObserver.kt:55:5)
    at 9   shared                              0x10a015bcb        kfun:androidx.compose.runtime.CompositionImpl.RememberEventDispatcher.dispatchRememberObservers#internal + 1819 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Composition.kt:1295:34)
    at 10  shared                              0x10a010b2f        kfun:androidx.compose.runtime.CompositionImpl.applyChangesInLocked#internal + 1371 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Composition.kt:984:21)
    at 11  shared                              0x10a01165b        kfun:androidx.compose.runtime.CompositionImpl#applyChanges(){} + 479 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Composition.kt:1005:17)
    at 12  shared                              0x10a12551f        kfun:androidx.compose.runtime.ControlledComposition#applyChanges(){}-trampoline + 91 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Composition.kt:254:5)
    at 13  shared                              0x10a036b8f        kfun:androidx.compose.runtime.Recomposer.runRecomposeAndApplyChanges$lambda$4$lambda$3#internal + 4751 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Recomposer.kt:639:45)
    at 14  shared                              0x10a03b0e3        kfun:androidx.compose.runtime.Recomposer.$runRecomposeAndApplyChanges$lambda$4$lambda$3$FUNCTION_REFERENCE$13.invoke#internal + 127 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Recomposer.kt:551:30)
    at 15  shared                              0x10a03b32b        kfun:androidx.compose.runtime.Recomposer.$runRecomposeAndApplyChanges$lambda$4$lambda$3$FUNCTION_REFERENCE$13.$<bridge-UNNB>invoke(kotlin.Long){}#internal + 123 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Recomposer.kt:551:30)
    at 16  shared                              0x109eb8047        kfun:kotlin.Function1#invoke(1:0){}1:1-trampoline + 107 (/Users/teamcity/.gradle/daemon/8.2.1/[K][Suspend]Functions:1:1)
    at 17  shared                              0x109fe205f        kfun:androidx.compose.runtime.BroadcastFrameClock.FrameAwaiter.resume#internal + 439 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/BroadcastFrameClock.kt:42:51)
    at 18  shared                              0x109fe2933        kfun:androidx.compose.runtime.BroadcastFrameClock#sendFrame(kotlin.Long){} + 499 (/opt/buildAgent/work/839471fc329135c7/compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/BroadcastFrameClock.kt:71:29)
    at 19  shared                              0x10aa9020f        kfun:androidx.compose.ui.scene.BaseComposeScene#render(androidx.compose.ui.graphics.Canvas;kotlin.Long){} + 731 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt:163:20)
    at 20  shared                              0x10ab5f523        kfun:androidx.compose.ui.scene.ComposeScene#render(androidx.compose.ui.graphics.Canvas;kotlin.Long){}-trampoline + 107 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/ComposeScene.skiko.kt:161:5)
    at 21  shared                              0x10aaf051b        kfun:androidx.compose.ui.scene.RenderingUIViewDelegateImpl.render#internal + 523 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt:155:15)
    at 22  shared                              0x10ab69db3        kfun:androidx.compose.ui.window.RenderingUIView.Delegate#render(org.jetbrains.skia.Canvas;kotlin.Double){}-trampoline + 107 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:46:9)
    at 23  shared                              0x10ab409b3        kfun:androidx.compose.ui.window.RenderingUIView.object-1.render#internal + 187 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:65:32)
    at 24  shared                              0x10ab69993        kfun:androidx.compose.ui.window.MetalRedrawerCallbacks#render(org.jetbrains.skia.Canvas;kotlin.Double){}-trampoline + 107 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:163:5)
    at 25  shared                              0x10ab3942f        kfun:androidx.compose.ui.window.MetalRedrawer.draw#internal + 2759 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:356:27)
    at 26  shared                              0x10ab388df        kfun:androidx.compose.ui.window.MetalRedrawer#drawSynchronously(){} + 203 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:329:9)
    at 27  shared                              0x10ab403c7        kfun:androidx.compose.ui.window.RenderingUIView.updateMetalLayerSize#internal + 4183 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:137:9)
    at 28  shared                              0x10ab3f2e7        kfun:androidx.compose.ui.window.RenderingUIView#objc:layoutSubviews + 215 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:118:31)
    at 29  shared                              0x10ab414d7        _6f72672e6a6574627261696e732e636f6d706f73652e75693a75692f6f70742f6275696c644167656e742f776f726b2f383339343731666333323931333563372f636f6d706f73652f75692f75692f7372632f75696b69744d61696e2f6b6f746c696e2f616e64726f6964782f636f6d706f73652f75692f77696e646f772f52656e646572696e675549566965772e75696b69742e6b74_knbridge70 + 143 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:117:14)
    at 30  UIKitCore                           0x185806d77        -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 1491 
    at 31  QuartzCore                          0x189ffa5af        _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 439 
    at 32  QuartzCore                          0x18a005367        _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 123 
    at 33  QuartzCore                          0x189f2051f        _ZN2CA7Context18commit_transactionEPNS_11TransactionEdPd + 459 
    at 34  QuartzCore                          0x189f4fcbf        _ZN2CA11Transaction6commitEv + 651 
    at 35  QuartzCore                          0x189f51193        _ZN2CA11Transaction25flush_as_runloop_observerEb + 67 
    at 36  CoreFoundation                      0x1803f0ec3        __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 31 
    at 37  CoreFoundation                      0x1803eb8c7        __CFRunLoopDoObservers + 527 
    at 38  CoreFoundation                      0x1803ebd7f        __CFRunLoopRun + 967 
    at 39  CoreFoundation                      0x1803eb5a3        CFRunLoopRunSpecific + 571 
    at 40  GraphicsServices                    0x18e9fbae3        GSEventRunModal + 159 
    at 41  UIKitCore                           0x1852f02e3        -[UIApplication _run] + 867 
    at 42  UIKitCore                           0x1852f3f5b        UIApplicationMain + 123 
    at 43  SwiftUI                             0x1c51fc1af        OUTLINED_FUNCTION_70 + 499 
    at 44  SwiftUI                             0x1c51fc04f        OUTLINED_FUNCTION_70 + 147 
    at 45  SwiftUI                             0x1c4f02fa3        OUTLINED_FUNCTION_2 + 91 
    at 46  My application                      0x10451c7f7        $s14My_application6iOSAppV5$mainyyFZ + 39 
    at 47  My application                      0x10451cff3        main + 11 (<userhome>/compose-cupertino/iosApp/iosApp/iOSApp.swift:<unknown>)
    at 48  dyld                                0x104ba1543        0x0 + 4374271299 
    at 49  ???                                 0x1045a20df        0x0 + 4367982815 
    at 50  ???                                 0xff57ffffffffffff 0x0 + -47287796087390209 
    Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [androidx.compose.runtime.BroadcastFrameClock@61113ff0, StandaloneCoroutine{Cancelling}@642549a0, FlushCoroutineDispatcher@52be550]
        at 0   shared                              0x109d98a77        kfun:kotlin.Throwable#<init>(kotlin.String?){} + 119 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Throwable.kt:28:37)
        at 1   shared                              0x109d91fe3        kfun:kotlin.Exception#<init>(kotlin.String?){} + 115 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:23:44)
        at 2   shared                              0x109d92203        kfun:kotlin.RuntimeException#<init>(kotlin.String?){} + 115 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/Exceptions.kt:34:44)
        at 3   shared                              0x109fd1b37        kfun:kotlinx.coroutines.internal.DiagnosticCoroutineContextException#<init>(kotlin.coroutines.CoroutineContext){} + 167 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/internal/CoroutineExceptionHandlerImpl.kt:27:5)
        at 4   shared                              0x109fa6903        kfun:kotlinx.coroutines.internal#handleUncaughtCoroutineException(kotlin.coroutines.CoroutineContext;kotlin.Throwable){} + 647 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/internal/CoroutineExceptionHandlerImpl.common.kt:43:33)
        at 5   shared                              0x109f49c73        kfun:kotlinx.coroutines#handleCoroutineException(kotlin.coroutines.CoroutineContext;kotlin.Throwable){} + 515 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/CoroutineExceptionHandler.kt:28:5)
        at 6   shared                              0x109f3df47        kfun:kotlinx.coroutines.StandaloneCoroutine.handleJobException#internal + 167 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/Builders.common.kt:190:9)
        at 7   shared                              0x109fd8b07        kfun:kotlinx.coroutines.JobSupport#handleJobException(kotlin.Throwable){}kotlin.Boolean-trampoline + 59 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/JobSupport.kt:1028:20)
        at 8   shared                              0x109f53617        kfun:kotlinx.coroutines.JobSupport.finalizeFinishingState#internal + 1135 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/JobSupport.kt:228:59)
        at 9   shared                              0x109f5b9b3        kfun:kotlinx.coroutines.JobSupport.tryMakeCompletingSlowPath#internal + 1983 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/JobSupport.kt:907:16)
        at 10  shared                              0x109f5b1bb        kfun:kotlinx.coroutines.JobSupport.tryMakeCompleting#internal + 755 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/JobSupport.kt:864:16)
        at 11  shared                              0x109f5aceb        kfun:kotlinx.coroutines.JobSupport#makeCompletingOnce(kotlin.Any?){}kotlin.Any? + 379 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/JobSupport.kt:829:30)
        at 12  shared                              0x109f3bed7        kfun:kotlinx.coroutines.AbstractCoroutine#resumeWith(kotlin.Result<1:0>){} + 211 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/AbstractCoroutine.kt:97:21)
        at 13  shared                              0x109ebcc6b        kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
        at 14  shared                              0x109d9d523        kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1163 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/coroutines/ContinuationImpl.kt:43:32)
        at 15  shared                              0x109ebcc6b        kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
        at 16  shared                              0x109fcae2b        kfun:kotlinx.coroutines.UndispatchedCoroutine#afterResume(kotlin.Any?){} + 243 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/CoroutineContext.kt:52:51)
        at 17  shared                              0x109fd553b        kfun:kotlinx.coroutines.AbstractCoroutine#afterResume(kotlin.Any?){}-trampoline + 59 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/AbstractCoroutine.kt:<unknown>)
        at 18  shared                              0x109f3bf33        kfun:kotlinx.coroutines.AbstractCoroutine#resumeWith(kotlin.Result<1:0>){} + 303 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/AbstractCoroutine.kt:99:9)
        at 19  shared                              0x109ebcc6b        kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
        at 20  shared                              0x109d9d523        kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1163 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/coroutines/ContinuationImpl.kt:43:32)
        at 21  shared                              0x109ebcc6b        kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
        at 22  shared                              0x109fb1443        kfun:kotlinx.coroutines.internal.ScopeCoroutine#afterResume(kotlin.Any?){} + 243 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/internal/Scopes.kt:28:15)
        at 23  shared                              0x109fd553b        kfun:kotlinx.coroutines.AbstractCoroutine#afterResume(kotlin.Any?){}-trampoline + 59 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/AbstractCoroutine.kt:<unknown>)
        at 24  shared                              0x109f3bf33        kfun:kotlinx.coroutines.AbstractCoroutine#resumeWith(kotlin.Result<1:0>){} + 303 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/AbstractCoroutine.kt:99:9)
        at 25  shared                              0x109ebcc6b        kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
        at 26  shared                              0x109d9d523        kfun:kotlin.coroutines.native.internal.BaseContinuationImpl#resumeWith(kotlin.Result<kotlin.Any?>){} + 1163 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/kotlin-native/runtime/src/main/kotlin/kotlin/coroutines/ContinuationImpl.kt:43:32)
        at 27  shared                              0x109ebcc6b        kfun:kotlin.coroutines.Continuation#resumeWith(kotlin.Result<1:0>){}-trampoline + 99 (/opt/buildAgent/work/2fed3917837e7e79/kotlin/libraries/stdlib/src/kotlin/coroutines/Continuation.kt:26:12)
        at 28  shared                              0x109faa59b        kfun:kotlinx.coroutines.DispatchedTask#run(){} + 1691 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/common/src/internal/DispatchedTask.kt:102:58)
        at 29  shared                              0x109fd7d97        kfun:kotlinx.coroutines.Runnable#run(){}-trampoline + 91 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Runnable.kt:10:19)
        at 30  shared                              0x10aa8013b        kfun:androidx.compose.ui.platform.FlushCoroutineDispatcher.flush$lambda$2#internal + 787 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt:88:14)
        at 31  shared                              0x10aa80d13        kfun:androidx.compose.ui.platform.FlushCoroutineDispatcher.$flush$lambda$2$FUNCTION_REFERENCE$1.invoke#internal + 71 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt:78:19)
        at 32  shared                              0x10aa80de3        kfun:androidx.compose.ui.platform.FlushCoroutineDispatcher.$flush$lambda$2$FUNCTION_REFERENCE$1.$<bridge-UNN>invoke(){}#internal + 71 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt:78:19)
        at 33  shared                              0x109eb95af        kfun:kotlin.Function0#invoke(){}1:0-trampoline + 99 (/Users/teamcity/.gradle/daemon/8.2.1/[K][Suspend]Functions:1:1)
        at 34  shared                              0x10aa7f603        kfun:androidx.compose.ui.platform.FlushCoroutineDispatcher.performRun#internal + 303 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt:99:13)
        at 35  shared                              0x10aa7f4af        kfun:androidx.compose.ui.platform.FlushCoroutineDispatcher#flush(){} + 163 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/FlushCoroutineDispatcher.skiko.kt:78:19)
        at 36  shared                              0x10aa97de3        kfun:androidx.compose.ui.scene.ComposeSceneRecomposer#performScheduledTasks(){} + 131 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/ComposeSceneRecomposer.skiko.kt:81:29)
        at 37  shared                              0x10aa901e7        kfun:androidx.compose.ui.scene.BaseComposeScene#render(androidx.compose.ui.graphics.Canvas;kotlin.Long){} + 691 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/BaseComposeScene.skiko.kt:162:20)
        at 38  shared                              0x10ab5f523        kfun:androidx.compose.ui.scene.ComposeScene#render(androidx.compose.ui.graphics.Canvas;kotlin.Long){}-trampoline + 107 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/ComposeScene.skiko.kt:161:5)
        at 39  shared                              0x10aaf051b        kfun:androidx.compose.ui.scene.RenderingUIViewDelegateImpl.render#internal + 523 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/scene/ComposeSceneMediator.uikit.kt:155:15)
        at 40  shared                              0x10ab69db3        kfun:androidx.compose.ui.window.RenderingUIView.Delegate#render(org.jetbrains.skia.Canvas;kotlin.Double){}-trampoline + 107 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:46:9)
        at 41  shared                              0x10ab409b3        kfun:androidx.compose.ui.window.RenderingUIView.object-1.render#internal + 187 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/RenderingUIView.uikit.kt:65:32)
        at 42  shared                              0x10ab69993        kfun:androidx.compose.ui.window.MetalRedrawerCallbacks#render(org.jetbrains.skia.Canvas;kotlin.Double){}-trampoline + 107 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:163:5)
        at 43  shared                              0x10ab3942f        kfun:androidx.compose.ui.window.MetalRedrawer.draw#internal + 2759 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:356:27)
        at 44  shared                              0x10ab3acfb        kfun:androidx.compose.ui.window.MetalRedrawer.<init>$lambda$0#internal + 435 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:263:17)
        at 45  shared                              0x10ab3be87        kfun:androidx.compose.ui.window.MetalRedrawer.$<init>$lambda$0$FUNCTION_REFERENCE$0.invoke#internal + 71 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:258:63)
        at 46  shared                              0x10ab3bf57        kfun:androidx.compose.ui.window.MetalRedrawer.$<init>$lambda$0$FUNCTION_REFERENCE$0.$<bridge-UNN>invoke(){}#internal + 71 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:258:63)
        at 47  shared                              0x109eb95af        kfun:kotlin.Function0#invoke(){}1:0-trampoline + 99 (/Users/teamcity/.gradle/daemon/8.2.1/[K][Suspend]Functions:1:1)
        at 48  shared                              0x10ab3cc9f        kfun:androidx.compose.ui.window.DisplayLinkProxy.handleDisplayLinkTick#internal + 151 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:515:9)
        at 49  shared                              0x10ab3cd53        kfun:androidx.compose.ui.window.DisplayLinkProxy.$imp:handleDisplayLinkTick#internal + 143 (/opt/buildAgent/work/839471fc329135c7/compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/MetalRedrawer.uikit.kt:514:5)
        at 50  QuartzCore                          0x189e4cf6f        _ZN2CA7Display15DisplayLinkItem9dispatch_ERNS_8SignPost8IntervalILNS2_11CAEventCodeE835322056EEE + 43 
        at 51  QuartzCore                          0x189e4e1a7        _ZN2CA7Display11DisplayLink14dispatch_itemsEyyy + 803 
        at 52  QuartzCore                          0x189e4dd73        _ZN2CA7Display11DisplayLink8callbackEP15_CADisplayTimeryyybPv + 631 
        at 53  QuartzCore                          0x189f54c8b        _ZL22display_timer_callbackP12__CFMachPortPvlS1_ + 335 
        at 54  CoreFoundation                      0x1803bbd1f        __CFMachPortPerform + 171 
        at 55  CoreFoundation                      0x1803f2657        __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 55 
        at 56  CoreFoundation                      0x1803f1c27        __CFRunLoopDoSource1 + 495 
        at 57  CoreFoundation                      0x1803ec237        __CFRunLoopRun + 2175 
        at 58  CoreFoundation                      0x1803eb5a3        CFRunLoopRunSpecific + 571 
        ... and 11 more common stack frames skipped
```

**Comment by alexzhirkevich** [comment]
Thanks. Crash is already fixed and will be available in next build. But NativeChildren currently has a memory leak (not on the library side). Prefer using compose variant of back gesture. In next release it will have better animation curve and duration and will 100% replicate native iOS navigation

**Comment by martin-reiterer** [comment]
Thanks for your fast reply! 

Do you mean (predictive) back gesture support of the `decompose` library? Or is there also something directly in compose, as you wrote? Now, I've tried it with decompose back gesture overlay and it seems to come very close to the native iOS navigation behavior. Only as a minor side-effect, there seems to be a short flicker of the Top-Bar title at the beginning of the animation.

**Comment by alexzhirkevich** [comment]
>Only as a minor side-effect, there seems to be a short flicker of the Top-Bar title at the beginning of the animation.

It doesn't seem right. Can you post a reproducer?

**Comment by martin-reiterer** [comment]
Following the instructions in [Decompose docs - Predictive Back Gesture on iOS](https://github.com/arkivanov/Decompose/blob/master/docs/extensions/compose.md#predictive-back-gesture-on-ios) I've added a custom ios-like back animation based on the current state of the `master` branch (6b4c8a4e22dfd81b9b75b4b438cbad3e4cb28784) of compose-cupertino. 

The changes can be applied using the following patch file: [Use_decompose_Children_Composable.patch](https://github.com/alexzhirkevich/compose-cupertino/files/14882368/Use_decompose_Children_Composable.patch)

When using this back animation, the Top-Bar title is displayed for a short time, directly at the beginning of the back navigation. 

<img width="368" alt="Bildschirmfoto 2024-04-05 um 10 32 29" src="https://github.com/alexzhirkevich/compose-cupertino/assets/11654016/ccd41058-161e-4ecd-bd1e-d8d1eadb0186">

**Comment by alexzhirkevich** [comment]
It is an implementation limitation of `NavigationTitle`  that will also be improved in the next patch. It would work well if you don't use that large collapsing title

**Comment by alexzhirkevich** [comment]
Fixed in `0.1.0-alpha04`
